fuzzy-native/src/MatcherBase.cpp

```diff
@@ -35,12 +35,51 @@ inline string str_to_lower(const std::string &s) {
   return lower;
 }
 
+float distance_to_root(const std::string *root, const std::string *value) {
+  size_t num_common_dirs = 0;
+  size_t i = 0;
+
+  // Count number of common directories
+  for (; i < root->length() && i < value->length(); i++) {
+    if (root->at(i) != value->at(i)) {
+      break;
+    }
+    if (root->at(i) == '/' || root->at(i) == '\\') {
+      num_common_dirs++;
+    }
+  }
+
+  if (i == root->length()) {
+    return 0;
+  }
+
+  // Count total number of directories in root
+  size_t num_dirs_root = num_common_dirs;
+  for (size_t j = i; j < root->length(); j++) {
+    if (root->at(j) == '/' || root->at(j) == '\\') {
+      num_dirs_root++;
+    }
+  }
+
+  float distance = num_dirs_root - num_common_dirs;
+  float penalty = 1;
+  for (size_t j = i; j < value->length(); j++) {
+    if (value->at(j) == '/' || value->at(j) == '\\') {
+      distance += penalty;
+      penalty *= 2.0;
+    }
+  }
+
+  return distance;
+}
+
 // Push a new entry on the heap while ensuring size <= max_results.
 void push_heap(ResultHeap &heap,
                float score,
+               float distance_to_root,
```

Comment:
It seems like all the computation above can fit in a plain `int`, so this should also be a plain int.